sdk/azidentity/managed_identity_client.go

```diff
@@ -120,43 +120,52 @@ func (c *managedIdentityClient) sendAuthRequest(ctx context.Context, msiType msi
 		return nil, err
 	}
 
+	// This should never happen under normal conditions
+	if resp == nil {
+		return nil, &AuthenticationFailedError{Err: errors.New("Something unexpected happened with the request and received a nil response")}
+	}
+
 	if resp.HasStatusCode(successStatusCodes[:]...) {
 		return c.createAccessToken(resp)
 	}
 
-	var authFailed AuthenticationFailedError
-	err = json.Unmarshal(resp.Payload, &authFailed)
-	if err != nil {
-		authFailed.Message = resp.Status
-		authFailed.Description = "Failed to unmarshal response: " + err.Error()
-	}
-	authFailed.Response = resp
-	return nil, &authFailed
+	return nil, &AuthenticationFailedError{Err: newAuthenticationResponseError(resp)}
 }
 
 func (c *managedIdentityClient) createAccessToken(res *azcore.Response) (*azcore.AccessToken, error) {
-	value := azcore.AccessToken{}
+	value := internalAccessToken{}
+	accessToken := &azcore.AccessToken{}
 	if err := json.Unmarshal(res.Payload, &value); err != nil {
-		return nil, fmt.Errorf("azcore.AccessToken: %w", err)
+		return nil, fmt.Errorf("internalAccessToken: %w", err)
+	}
+
+	if value.ExpiresIn == "" {
+		if value.ExpiresOn == "" {
+			return nil, &AuthenticationFailedError{Err: errors.New("did not receive a valid token expiration time")}
+		}
+		accessToken.Token = value.Token
+		// TODO: missing here is parsing expires on to a time.Time value
```

Comment:
Does this TODO need to be addressed now?

Reply:
It has been addressed, but it'll be merged in one of my next PRs.